### ISSUE TITLE
whereEval does not work

### ISSUE THREAD

**Issue by darshan-scaletech** [phulst/node-tile38#28]
Used whereEval to run a script **("return FIELDS.specialtily > ARGV[1]", 1)** which queries on custom fields, but I keep getting syntax error

**Comment by merewolf** [comment]
@phulst hi, I have the same issue. Any plans for fixing it?

**Comment by phulst** [comment]
can you share some example code that reproduces this error? This will help me identify the root cause and fix.

**Comment by merewolf** [comment]
Hi, so if I do a request from the cli
`NEARBY area LIMIT 1 DISTANCE WHEREEVAL "return FIELDS.areaId ~= ARGV[1]" 1 111631 POINT 16.3063988578379 48.1598374510414 3500`
Search works correctly and I get a response:
`{"ok":true,"fields":["areaId"],"objects":[{"id":"d240b2ecaa35b03a","object":{"type":"Point","coordinates":[48.1598374510414,16.3063988578379]},"fields":[111631],"distance":0}],"count":1,"cursor":1,"elapsed":"68.014µs"}`

however in my node app I have a call
```
let searchQuery = client.nearbyQuery('area')
            .distance()
            .point(req.body.long, req.body.lat, req.body.maxDistance)
            .limit(1)
            .whereEval("return FIELDS.areaId ~= ARGV[1]", 111631)

        searchResult = await searchQuery.execute()
```
And the response is:
```
sending command "NEARBY area LIMIT 1 DISTANCE WHEREEVAL "return FIELDS.areaId ~= ARGV[1]" 1 111631 POINT 16.3063988578379 48.1598374510414 3500"
{"ok":false,"err":"f_0dd6a596e4940107b7c0aea39dfb780ef134d1df line:1(column:33) near 'return FIELDS.areaId ~= ARGV[1]':   syntax error\\n","elapsed":"75.654µs"}

```
As you can see, the command that is logged is identical to the one that is working in cli, but for some reason the response is an error when using node client.
I hope this is enough info for you to go on, let me know if you need more.

**Comment by phulst** [comment]
interesting... this is very helpful. I'll try it out and will let you know what I find. Does the WHEREEVAL command work fine for you with other values?

**Comment by merewolf** [comment]
All of my data is inserted with this syntax
`SET area 9621a40b0f90f467 FIELD areaId 12345 POINT 13.3252703267768 52.5029839702301`
and all of them throw the same error.
In client I parse a large file andthe command looks like
`client.set('area', uid.uid(16), point, {areaId: areaId});
`
where point is the coordinates and areaId is a field

**Comment by phulst** [comment]
sorry just now doing some fixes and working on a new revision. I was able to reproduce this error, but the Syntax error is returned by the Tile38 server, and the node driver doesn't manipulate the script (or response) in any way. I have no reason to believe that it's passing your commands incorrectly to the server, so I suspect this is a server bug or that the CLI works slightly different than the API for some reason. I've opened up a bug with Tile38, hopefully the maintainer can shed some light on why it isn't working.
